### ISSUE TITLE
Multiple Transition run in sequence instead of simultaneously

### ISSUE THREAD

**Issue by hburrows** [reactjs/react-transition-group#275]
I want 2 separate `Transition` components to transition simultaneously in parallel (this is a follow on to a question I posted on stackoverflow a couple days ago https://stackoverflow.com/questions/48235984/reacttransitiongroup-simultaneous-transitions).  When I express my effect via `Transition` components they execute one after the other.  However, if I re-write this to not use `Transition` components and just use `div`s and regular CSS it works fine.  I suspect `Transition` components flush their updates in separate render cycles which causes this sequencing.  Is there a way to cause multiple independent transitions to render in the same cycle using different `Transition` components (that may or may not be rendering inside a `TransitionGroup`)?

This sequential transitioning behavior happens on all browsers that I've tested (chrome, safari, and firefox)

**Comment by jquense** [comment]
Can you provide a runnable example of it not working?

**Comment by jquense** [comment]
Ok the issue is probably that you need to manually reflow the browser to make sure the transitions start at the right time. there are a bunch of previous issues on the topic, but generally you use a hook and the node reference to trigger a reflow onEnter or onEntering

**Comment by hburrows** [comment]
@jquense Thanks for the quick response.  I'm not totally clear what you mean by "manually reflow the browser".  How do I trigger that?

Also, would the behavior change if I switched to using `CSSTransition` instead of `Transition`?  It would be easy enough to refactor.

**Comment by jquense** [comment]
> Also, would the behavior change if I switched to using CSSTransition instead of Transition?

Probably, its higher level and handles some of these cases.

**Comment by hburrows** [comment]
Yeah.... switching to `CSSTransition` fixes the problem.  Can you provide me with a pointer or two on how to "manually reflow the browser" so if I need to use `Transition` in the future I know the tricks.  Googling "reflow browser" doesn't reveal much -- everything is focus on how to minimize browser reflow.

**Comment by jquense** [comment]
check out: https://github.com/reactjs/react-transition-group/blob/master/src/CSSTransition.js#L208-L212
